test/rulesets/Base/mapreduce.jl

```diff
@@ -88,6 +88,14 @@
         test_rrule(sum, sqrt, randn(5,5) .> 0; fkwargs=(;dims=1))
         # ... and Bool produced by function
         @test_skip test_rrule(sum, iszero, randn(5))  # DimensionMismatch("second dimension of A, 1, does not match length of x, 0")
+
+
+        # Functions that return a Vector
+        # see https://github.com/FluxML/Zygote.jl/issues/1074
+        test_rrule(sum, make_two_vec, [1.0, 3.0, 5.0, 7.0])
+        test_rrule(sum, make_two_vec, [1.0 2.0; 3.0 4.0])
+        test_rrule(sum, make_two_vec, [1.0 2.0; 3.0 4.0]; fkwargs=(;dims=2))
+        test_rrule(sum, make_two_vec, [1.0 2.0; 3.0 4.0]; fkwargs=(;dims=1))
```

Comment:
Maybe a weirder test:
```suggestion
        test_rrule(sum, make_two_vec, [1.0 2.0; 3.0 4.0]; fkwargs=(;dims=(3,4)))
```
Maybe also worth testing something in which the gradient is in the function not the array, like `gradient(x -> sum(sum(Returns([x,1]), 1:3)), 4.0)`. Maybe it would handle this?
```
gradient(x -> sum(sum(Multiplier([x 0; 0 1]), [[1 2; 3 4], [5 6; 7 8]])), 4.0)
```
I tried `gradient(x -> sum(sum(Multiplier([x 0; 0 1]), [[1,2], [3,4]])), 4.0)` and got an error, not sure if that ought to work.

Reply:
Not sure about the ones where the gradient is in the function.
In particular not sure how to add them without needing an AD system.
(Maybe we should write a RuleConfig for FiniteDifferences.jl and use that in CRTU...)